internal/telemetry/metrics.go

```diff
@@ -57,9 +50,42 @@ func RegisterTelemetryCollector(datastoreEngine string, ds datastore.Datastore)
 				"arch":       runtime.GOARCH,
 				"go":         buildInfo.GoVersion,
```

Comment:
we are already providing go version through the standard Go extended runtime metrics. It's available through `go_info`

Reply:
Yeah, this is an artifact of reusing the same metric across telemetry and non-telemetry usage. Should I make it optional or just keep it?

Reply:
it's ok, I don't think it hurts, just wanted to call it out